spec/factories/document_type_factory.rb

```diff
@@ -11,11 +11,15 @@
 
     id { SecureRandom.hex(4) }
     label { SecureRandom.alphanumeric(8) }
-    contents { [] }
     tags { [] }
     guidance { [] }
     topics { false }
 
+    contents do
+      [DocumentType::TitleAndBasePathField.new,
+       DocumentType::SummaryField.new]
+    end
```

Comment:
I'm not sure what you mean here; could you help me understand? Since the purpose of the factory is to create an instance of the object, it seemed naturally to use its constructor directly.

Reply:
Yes sorry. It is fine to initialise the classes in the factory. 

I was referring to how we're having to pass in classes in the tests: https://github.com/alphagov/content-publisher/blob/720412706763e62fe936f1fdc738100beb946a14/spec/lib/whitehall_importer/integrity_checker_spec.rb#L64-L69

and thought it'd be nicer if the call to the factory could be more like

```
build(:document_type, content_fields: %w[title_and_base_path summary body])
```

where the factory could convert those strings in the same way YAML does to get the classes?

Reply:
Ah, I see. That would be a little easier. I have a point on the card to try and address this with a `newsesque` trait, since we generally have (or don't care about) all three fields together. If we go with the trait option, we could add support for field IDs in future too. Perhaps we can discuss this further on the subsequent PR I'm intending to raise for this?

Reply:
Yeah, no problem for dealing with on a later PR.

I think the closer we can have this configuration to what's in the YAML the better by the way, be even better if the building of the fields could be done by the same code that does the building of the YAML config.

Reply:
I think we used to do that sort of thing for classes like `DocumentType`, but I remember it made them harder to work with in tests, probably because the generation of the transient hash wasn't very intuitive, given the layer of indirection it represents.

Found the commit that describes the pain: https://github.com/alphagov/content-publisher/pull/622/commits/24b4fd21805c3edd94ab4b64a675c7cde29736ee

I like the random "foo" at the end 😆